backpack/extensions/backprop_extension.py

```diff
@@ -74,38 +89,56 @@ def set_module_extension(
             )
         self.__module_extensions[module] = extension
 
-    def __get_module_extension(self, module):
+    def __get_module_extension(
+        self, module: Module
+    ) -> Callable[[BackpropExtension, Module, Tuple[Tensor], Tuple[Tensor]], None]:
         module_extension = self.__module_extensions.get(module.__class__)
 
         if module_extension is None:
-
-            if isinstance(module, (Sequential, ReduceTuple)):
-                return no_op
-
             if self.__fail_mode is FAIL_ERROR:
                 raise NotImplementedError(
-                    "Extension saving to {} ".format(self.savefield)
-                    + "does not have an extension for "
-                    + "Module {}".format(module.__class__)
+                    f"Extension saving to {self.savefield} "
+                    "does not have an extension for "
+                    f"Module {module.__class__}"
                 )
             elif self.__fail_mode == FAIL_WARN:
-                warnings.warn(
-                    "Extension saving to {} ".format(self.savefield)
-                    + "does not have an extension for "
-                    + "Module {}".format(module.__class__)
-                )
+                for _ in module.parameters():
+                    warnings.warn(
+                        f"Extension saving to {self.savefield} does not have an "
+                        f"extension for Module {module.__class__} "
+                        f"although the module has parameters"
+                    )
+                    break
 
-            return no_op
+            return no_op_apply
 
         return module_extension.apply
 
-    def apply(self, module, g_inp, g_out):
+    def apply(
+        self,
+        module: Module,
+        g_inp: Tuple[Tensor],
+        g_out: Tuple[Tensor],
+    ) -> None:
         """Applies backpropagation.
 
         Args:
-            module(torch.nn.module): module to perform backpropagation on
-            g_inp(tuple[torch.Tensor]): input gradient
-            g_out(tuple[torch.Tensor]): output gradient
+            module: module to perform backpropagation on
+            g_inp: input gradient
+            g_out: output gradient
         """
         module_extension = self.__get_module_extension(module)
         module_extension(self, module, g_inp, g_out)
+
+    @abc.abstractmethod
+    def expects_backpropagation_quantities(self) -> bool:
+        """Whether the extension uses additional backpropagation quantities.
+
+        Returns:
+            Whether the extension uses additional backpropagation quantities.
+        """
+        return
```

Comment:
`pass` instead?

Reply:
I'm not sure. If I use `pass`, darglint complains about an excess `Returns` in the docstring. I would keep the `return`.

Reply:
Okay.